src/docker.rs

```diff
@@ -98,145 +193,412 @@ fn parse_docker_opts(value: &str) -> Result<Vec<String>> {
+}
 
-    let home_dir = home::home_dir().ok_or_else(|| eyre::eyre!("could not find home directory"))?;
-    let cargo_dir = home::cargo_home()?;
-    let xargo_dir = env::var_os("XARGO_HOME")
-        .map(PathBuf::from)
-        .unwrap_or_else(|| home_dir.join(".xargo"));
-    let nix_store_dir = env::var_os("NIX_STORE").map(PathBuf::from);
-    let target_dir = &metadata.target_directory;
-
-    // create the directories we are going to mount before we mount them,
-    // otherwise `docker` will create them but they will be owned by `root`
-    fs::create_dir(&target_dir).ok();
-    fs::create_dir(&cargo_dir).ok();
-    fs::create_dir(&xargo_dir).ok();
-
-    // update paths to the host mounts path.
-    let cargo_dir = mount_finder.find_mount_path(cargo_dir);
-    let xargo_dir = mount_finder.find_mount_path(xargo_dir);
-    let target_dir = mount_finder.find_mount_path(target_dir);
-    // root is either workspace_root, or, if we're outside the workspace root, the current directory
-    let host_root = mount_finder.find_mount_path(if metadata.workspace_root.starts_with(cwd) {
-        cwd
-    } else {
-        &metadata.workspace_root
-    });
-    let mount_root: PathBuf;
-    #[cfg(target_os = "windows")]
-    {
-        // On Windows, we can not mount the directory name directly. Instead, we use wslpath to convert the path to a linux compatible path.
-        mount_root = wslpath(&host_root, verbose)?;
-    }
-    #[cfg(not(target_os = "windows"))]
-    {
-        mount_root = mount_finder.find_mount_path(host_root.clone());
-    }
-    let mount_cwd: PathBuf;
-    #[cfg(target_os = "windows")]
-    {
-        // On Windows, we can not mount the directory name directly. Instead, we use wslpath to convert the path to a linux compatible path.
-        mount_cwd = wslpath(cwd, verbose)?;
-    }
-    #[cfg(not(target_os = "windows"))]
-    {
-        mount_cwd = mount_finder.find_mount_path(cwd);
+// copy files for a docker volume, for remote host support
+fn copy_volume_files(
+    engine: &Engine,
+    container: &str,
+    src: &Path,
+    dst: &Path,
+    verbose: bool,
+) -> Result<ExitStatus> {
+    docker_subcommand(engine, "cp")
+        .arg("-a")
+        .arg(&src.display().to_string())
+        .arg(format!("{container}:{}", dst.display()))
+        .run_and_get_status(verbose)
+}
+
+pub fn copy_volume_xargo(
+    engine: &Engine,
+    container: &str,
+    xargo_dir: &Path,
+    target: &Target,
+    mount_prefix: &Path,
+    verbose: bool,
+) -> Result<()> {
+    // only need to copy the rustlib files for our current target.
+    let triple = target.triple();
+    let relpath = Path::new("lib").join("rustlib").join(&triple);
+    let src = xargo_dir.join(&relpath);
+    let dst = mount_prefix.join("xargo").join(&relpath);
+    if Path::new(&src).exists() {
+        create_volume_dir(engine, container, dst.parent().unwrap(), verbose)?;
+        copy_volume_files(engine, container, &src, &dst, verbose)?;
     }
```

Comment:
why are there their own functions? can they not be summarized to one?

Reply:
> why are these their own functions? can they not be summarized to one?

Sure they could, but they're also somewhat different functionality each so I figured it would be handy (especially during testing) the separate them.

Reply:
resolved for now, will revisit when this is more fleshed out.

do you want to convert this to a draft for now maybe to signal clearly it's in progress?